detox/e2e/test/performance/login.e2e.ts

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// *******************************************************************
+// - [#] indicates a test step (e.g. # Go to a screen)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element testID when selecting an element. Create one if none.
+// *******************************************************************
+
+import fs from 'fs';
+
+import {Setup} from '@support/server_api';
+import client from '@support/server_api/client';
+import {
+    serverOneUrl,
+    siteOneUrl,
+} from '@support/test_config';
+import {
+    HomeScreen,
+    LoginScreen,
+    ServerScreen,
+} from '@support/ui/screen';
+import {computeStatsFromData} from '@support/utils/statistics';
+
+const N_RUNS = 10;
+describe('Performance test', () => {
+    let user: any;
+    const serverOneDisplayName = 'Server 1';
+    const times: Array<{prev: number; afterTap: number; after: number}> = [];
+
+    beforeAll(async () => {
+        const init = await Setup.apiInit(siteOneUrl);
+        user = init.user;
+
+        await ServerScreen.toBeVisible();
+        await ServerScreen.serverUrlInput.replaceText(serverOneUrl);
+        await ServerScreen.serverDisplayNameInput.replaceText(serverOneDisplayName);
+        client.post(`${process.env.REACT_APP_PERFORMANCE_WEBHOOK}/start_measures`, {measureId: 'login'});
+    });
+
+    beforeEach(async () => {
+        await ServerScreen.connectButton.tap();
+        await LoginScreen.toBeVisible();
+        await LoginScreen.usernameInput.replaceText(user.username);
+        await LoginScreen.passwordInput.replaceText(user.password);
+    });
+
+    afterEach(async () => {
+        await HomeScreen.logout();
+    });
+
+    afterAll(async () => {
+        const res = await client.post(`${process.env.REACT_APP_PERFORMANCE_WEBHOOK}/finish_measures`, {measureId: 'login'});
+        const calc = [];
+        for (let i = 0; i < N_RUNS; i++) {
+            calc.push({
+                total: times[i]!.after - res.data.measures.start[i],
+                totalDevice: res.data.measures.end[i] - res.data.measures.start[i],
+                testDelay: res.data.measures.start[i] - times[i]!.prev,
+                tapDelay: times[i]!.afterTap - times[i]!.prev,
+                testTotal: times[i]!.after - times[i]!.prev,
+                testAfterTap: times[i]!.after - times[i]!.afterTap,
+            });
```

Comment:
I am getting a few more measures mainly for testing. The important one (the one actually used) is `totalDevice`.